tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/CachedBasicPropertiesHelper.cs

```diff
@@ -1,23 +1,43 @@
-﻿// <copyright file="CachedBasicPropertiesHelper.cs" company="Datadog">
+// <copyright file="CachedBasicPropertiesHelper.cs" company="Datadog">
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
 #nullable enable
+using System;
+using System.Reflection.Emit;
+using System.Runtime.InteropServices;
+using Datadog.Trace.DuckTyping;
 using Datadog.Trace.Util;
 
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.RabbitMQ;
 
 internal static class CachedBasicPropertiesHelper<TBasicProperties>
 {
     // ReSharper disable once StaticMemberInGenericType
-    private static readonly ActivatorHelper Activator;
+    private static readonly Func<object, object> Activator;
 
     static CachedBasicPropertiesHelper()
     {
-        Activator = new ActivatorHelper(typeof(TBasicProperties).Assembly.GetType("RabbitMQ.Client.BasicProperties")!);
+        var targetType = typeof(TBasicProperties).Assembly.GetType("RabbitMQ.Client.BasicProperties")!;
+        var parameterType = typeof(TBasicProperties).Assembly.GetType("RabbitMQ.Client.IReadOnlyBasicProperties")!;
+
+        var constructor = targetType.GetConstructor([parameterType])!;
+
+        var createBasicPropertiesMethod = new DynamicMethod(
+            $"TypeActivator_{targetType.Name}_{parameterType.Name}",
+            targetType,
+            [parameterType],
+            typeof(DuckType).Module,
+            true);
+
+        var il = createBasicPropertiesMethod.GetILGenerator();
+        il.Emit(OpCodes.Ldarg_0);
+        il.Emit(OpCodes.Newobj, constructor);
+        il.Emit(OpCodes.Ret);
+        Activator = (Func<object, object>)createBasicPropertiesMethod.CreateDelegate(typeof(Func<object, object>), targetType);
```

Comment:
Unfortunately it looks like this is throwing an exception when attempting to run the tests / sample

```
[ERR] Exception occurred when calling the CallTarget integration continuation. System.TypeInitializationException: The type initializer for 'Datadog.Trace.ClrProfiler.AutoInstrumentation.RabbitMQ.CachedBasicPropertiesHelper`1' threw an exception.
 ---> System.ArgumentException: Cannot bind to the target method because its signature is not compatible with that of the delegate type.
   at System.Delegate.CreateDelegateNoSecurityCheck(Type type, Object target, RuntimeMethodHandle method)
   at System.Reflection.Emit.DynamicMethod.CreateDelegate(Type delegateType, Object target)
   at Datadog.Trace.ClrProfiler.AutoInstrumentation.RabbitMQ.CachedBasicPropertiesHelper`1..cctor() 
```